website/www/site/content/en/blog/dyi-content-discovery-platform-genai-beam.md

```diff
@@ -0,0 +1,338 @@
+  …
+}
+
+$ > curl -X POST \
+ -H "Content-Type: application/json" \
+ -H "Authorization: Bearer $(gcloud auth print-identity-token)" \
+ https://<service-address>/query/content \
+ -d $'{"text":"care to share the price?", "sessionId": "some-session-id"}' \
+ | jq .
+
+# response from service
+{
+  "content": "The VertexAI Foundational Models are priced based on the number of requests per second and the total number of requests per day. For more information, please see the [VertexAI Foundational Models pricing page](https://cloud.google.com/vertex-ai/pricing#foundational-models).",
+  …
+}
+```
+
+**Usage Tip:** in case of abruptly changing topics, sometimes is better to use a new session identifier.
+
+### Deployment
+
+As part of the platform solution, there are a set of scripts that help with the deployment of all the different components. By running the `start.sh` and setting the right parameters (GCP project, terraform state bucket and name for the platform instance) the script will take care of building the code, deploying the needed containers (service endpoint container and Dataflow python custom container), deploying all the GCP resources using Terraform and finally deploying the pipeline. There is also the possibility of modifying the pipeline’s execution by passing an extra parameter to the startup script, for example: `start.sh <gcp project> <state-bucket-name> <a run name> "--update"` will update the content extraction pipeline in-place.
+
+Also, in case of wanting to focus only on the deployment of specific components other scripts have been included to help with those specific tasks (build the solution, deploy the infrastructure, deploy the pipeline, deploy the services, etc.).
+
+### Solution's Notes
+
+This solution is designed to serve as an example for learning purposes. Many of the configuration values for the extraction pipeline and security restrictions are provided only as examples. The solution doesn't propagate the existing access control lists (ACLs) of the ingested content. As a result, all users that have access to the service endpoints have access to summarizations of the ingested content from those original documents.
+
+### Notes about the source code
+
+The source code for the content discovery platform is available in [Github](https://github.com/prodriguezdefino/content-dicovery-platform-gcp). You can run it in any Google Cloud project. The repository includes the source code for the integration services, the multi-language ingestion pipeline, and the deployment automation through Terraform. If you deploy this example, it might take up to 90 minutes to create and configure all the needed resources. The README file contains additional documentation about the deployment prerequisites and example REST interactions.
```

Comment:
Is it ok to point to personal github repositories? should this code be included in another place?

Reply:
That is ok.  I assume that you intent to keep it there. If you prefer, you can also move it to (https://github.com/GoogleCloudPlatform/dataflow-sample-applications). If you do that, please send another PR to update this blog post.

Reply:
Yes, I will try that out on a diff PR, thanks for the clarification.